receiver/httpcheckreceiver/metadata.yaml

```diff
@@ -54,3 +54,12 @@ metrics:
       monotonic: false
     unit: "{error}"
     attributes: [http.url, error.message]
+  httpcheck.substring:
```

Comment:
```suggestion
  httpcheck.body:
```

Reply:
> I wonder if a new metric is the right semantic here, or if instead it should be an attribute of an existing metric.

@codeboten this is a very good question. As a user, my assumption would be that if I specified a body match string in a target's configuration, then I only get `httpcheck.status` metrics with the value `1` if the body matched, and I'd expect getting `httpcheck.status` with value `0` and possibly an additional attribute e.g. `reason=body` or similar, explaining why the check failed.

I'm not sure how this fits the current design of the receiver, in that it reports a separate data point for each `http.status_class` (1xx, 2xx, etc.). By the way, I'm not sure what advantages this has over reporting just a single `httpcheck.status` data point for each target and request.

Reply:
Thanks @sakulali for all the changes. I still think the HTTP Check receiver should only emit a single metric, and a single data point for every target and check (HTTP request). This pull request adds another metric `httpcheck.body` to the already existing metric `httpcheck.status` that has multiple data points (one for every HTTP response code class - 1xx, 2xx, etc.).

This is not the direction I want to see for this receiver (see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22994), therefore I'm not going to approve this pull request. I'm also not going to stop this pull request from being merged if others approve it.

Reply:
Thanks for your detailed response @astencel-sumo! After reviewing the issue you mentioned, maybe i understand your concerns. I have a few thoughts and a small question:

1. Regarding the httpcheck.status metric, if there is only one data point, it seems that the value is always 1. In that case, how can we track the count of response codes in the 1xx, 2xx, 3xx, etc range?

2. If we don't add a separate metric `httpcheck.body`, instead use the body match as a condition for `httpcheck.status`, then the code will need further adjustments.

3. If we decide to keep only the httpcheck.status metric with a default single data point, I believe we would need a feature gate to enable or disable this functionality.

Since I'm not very familiar with the design of the HTTP Checker and not quite clear about the specific user scenarios, i would like to reach a consensus on a reasonable solution.

Reply:
I also believe that this can be an attribute of the result.
Putting it an example I think it's better for understandind.
Like if you are looking for something like a matching version on the target
the httpcheck.status = 200, but the httpcheck.body = false 
But when the version is updated, then the httpcheck.body = true.